### PR TITLE
feat: implement a basic async rest method

### PR DESCRIPTION
**Pull request by ohmayr**
This PR implements the `__call__` method for a basic asynchronous rest method and turns on the relevant unit tests to add coverage.

This PR should be reviewed and merged after: https://github.com/googleapis/gapic-generator-python/pull/2144.

### CODE REVIEW

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest_asyncio.py.j2 [hunk 1]
```diff
@@ -108,16 +112,62 @@ class Async{{service.name}}RestTransport(_Base{{ service.name }}RestTransport):
         def __hash__(self):
             return hash("Async{{service.name}}RestTransport.{{method.name}}")
 
+        {% if method.http_options and not method.client_streaming and not method.server_streaming and not method.lro and not method.extended_lro and not method.paged_result_field %}
```
Comment (parthea) [inline gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest_asyncio.py.j2:115]: Include TODO bug to follow up on removing this condition

**Review by parthea — APPROVED** [review]

File: tests/integration/goldens/redis/google/cloud/redis_v1/services/cloud_redis/transports/rest_asyncio.py [hunk 2]
```diff
@@ -250,29 +254,149 @@ class _GetInstance(_BaseCloudRedisRestTransport._BaseGetInstance, AsyncCloudRedi
...
+        @staticmethod
+        async def _get_response(
+            host,
+            metadata,
+            query_params,
+            session,
+            timeout,
+            transcoded_request,
+            body=None):
+
```
Comment (parthea) [inline tests/integration/goldens/redis/google/cloud/redis_v1/services/cloud_redis/transports/rest_asyncio.py:266]: Add type hints and docstring for this method

**Review by vchudnov-g — APPROVED** [review]

File: tests/integration/goldens/redis/google/cloud/redis_v1/services/cloud_redis/transports/rest_asyncio.py (hunk 2 above)
Comment (ohmayr) [reply tests/integration/goldens/redis/google/cloud/redis_v1/services/cloud_redis/transports/rest_asyncio.py:267]: This is a protected helper which pre-existed and is not a part of this PR's changed. I see that it's now also leveraged for async rest. Regardless, we can address this as a follow up. Let me know if that sounds good!

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest_asyncio.py.j2 (hunk 1 above)
Comment (ohmayr) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest_asyncio.py.j2:120]: addressed.

**Review by parthea — APPROVED** [review]

**Pull request merged**